### ISSUE TITLE
Expanded community integration testing

### ISSUE THREAD

**Issue by grampelberg** [linkerd/linkerd2#3900]
### What problem are you trying to solve?

As of today, the build and kind integration tests do not run for forks. This ends up creating issues for contributors when someone forgets to run the tests manually. Running tests automatically for forks can be a large security hole, so there needs to be some oversight to how these run.

### How should the problem be solved?

Trigger the full set of tests by adding a label to the PR. This will allow reviewers to do an initial review of a PR, add the `test/full` label to the PR and then get the results of the entire set.

### Any alternatives you've considered?

- Add the contributor to the Linkerd organization - there's a lot of friction here trying to get the contributor to agree to be added.
- Have a `/runtest` command - there's no way to visually see that the full tests have been run for this PR.
- Add to a testing project - cleanup's going to eventually be a problem.

**Comment by siggy** [comment]
Another option: For forks, run the integration tests directly on the GitHub Action box, it will be slower, but should not require any of the secrets.

**Comment by grampelberg** [comment]
@siggy WDYT works the best from a workflow perspective?

**Comment by siggy** [comment]
@grampelberg Short answer: there are tradeoffs...
1. Run forked integration tests directly on GitHub Actions box. If this works, it's the least effort, and ensures integration tests are run immediately on every PR. Integration tests would also be slower for forked PRs. It would involve not doing the `Docker SSH setup` and not setting `DOCKER_HOST` for forked PRs. We'd probably also need to set `max-parallel: 1` to avoid overloading the GitHub Actions box. 
2. Add a `test/full` label.  This would make integration tests fast for forked PRs. We'd have to build something that does this. I'm not aware of a way to give forked PRs elevated privs (secrets). Prow does this sort of thing. There's a GitHub Actions discussion here (it says "Solved", but its not): https://github.community/t5/GitHub-Actions/Make-secrets-available-to-builds-of-forks/td-p/30678
3. `/runtest`. Same as option 2.
4. Add contributor to Linkerd org. No code changes, but more work on the human process side. It would require external devs to move their PRs into the linkerd2 repo.

In summary: I think we should try option 1 as it's the clearest path to enabling integration tests for every PR. If that fails we should re-evaluate what other projects are doing.

**Comment by zaharidichev** [comment]
FWIW I am certainly a supporter for trying the first viable option (1) and see how it goes.

**Comment by alpeb** [comment]
Addressed via #4002
